test/vhswish-microkernel-tester.h

```diff
@@ -119,8 +119,8 @@ class VHSwishMicrokernelTester {
         const int32_t input_value = int32_t(uint32_t(input_zero_point() - input[i]) << 7);
         int32_t in = input_value * input_scale_div;
         in -= 16384;  // subtract 0.5 in Q15
-        in = std::min(in, 0);
-        in = std::max(in, -32768);
+        in = std::min<int32_t>(in, 0);
+        in = std::max<int32_t>(in, -32768);
```

Comment:
I see about 600 occurances of  std::min or std::max that dont specify a type.  Is int 32 bits?

Reply:
Yes it's int 32bits and yes I usually didn't need to specify the type, but I thought specifying type would fix the issue based on the error message I got:
```bash
/.../XNNPACK/test/vhswish-microkernel-tester.h:123:14: error: no matching function for call to 'max'
        in = std::max(in, -32768);
             ^~~~~~~~
/.../Hexagon_SDK/5.3.0.0/tools/HEXAGON_Tools/8.6.06/Tools/bin/../target/hexagon/include/c++/v1/__algorithm/max.h:39:1: note: candidate template ignored: deduced conflicting types for parameter '_Tp' ('long' vs. 'int')
max(const _Tp& __a, const _Tp& __b)
```
I assume this min/max implementations come from Hexagon SDK, which are targeted for Hexagon. Do you see other min/max are also using namespace std?

Reply:
This is definitely something weird with this compiler/runtime; either integer literals are of type `long` (which seems very unlikely) or their `int32_t` is an alias for `long` (which is plausible). FWIW, an alternate way to fix it is

```
        constexpr int32_t zero = 0;
        constexpr int32_t most_neg = -32768;
        in = std::min(in, zero);
        in = std::max(in, most_neg);
```

but this isn't really any more appealing.

Reply:
Yeah, `int32_t` is an alias for `long` rather than `int` in this environment, which is weird but legal I guess:

```
        static_assert(std::is_same<int32_t, int>::value, "int");  <-- fails
        static_assert(std::is_same<int32_t, long>::value, "long");  <-- passes
```

Since integer literals are all of type `int`, std::min that mixes an integer literal is mixing an `int` and a `long`, hence the error. The right way to fix this is to go back in time and change the definition of `int32_t` to be plain `int` here (since `int` and `long` are the same size here according to a similar static_assert() check) but that would be a nightmare to change now I presume, if for no other reason than it would require a new rev to the Hexagon SDK, I suspect.

TL;DR: adding the casts is probably the simplest way to address this.